oat-sa/extension-tao-testqti
# Fix/ADF-1789/sync items translations

### Related to: https://oat-sa.atlassian.net/browse/ADF-1789

Requires: 
 - [x] https://github.com/oat-sa/tao-core/pull/4122

### Summary

Synchronize the translation with the original upon opening a test for authoring

### Details

The call to the sync request must be made before loading the test model.

### How to test

- check out the branch: `git checkout -t origin/fix/ADF-1789/sync-items-translations`
- make sure to have the companion branch checked out and up to date too on the other extensions: `feat/HKD-6/integration`
- make sure to have the Feature Flags activated
```
FEATURE_FLAG_TRANSLATION_ENABLED=true
FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER=true
```
- make sure to have the Item Translator role assigned to your user
- modify the original test (add/remove an item)
- open a test translation: a feedback message is briefly shown. In the console, a `sync` request is sent to the translation API
- the modification must be visible in the translation (however, the item title may be missing)
